x/mint/keeper/mint.go

```diff
@@ -0,0 +1,53 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/dymensionxyz/dymension-rdk/x/mint/types"
+)
+
+func (k Keeper) HandleMintingEpoch(ctx sdk.Context) (sdk.Coins, error) {
+	var mintedCoins sdk.Coins
+	params := k.GetParams(ctx)
+
+	//calculate coins
+	total := k.bankKeeper.GetSupply(ctx, params.MintDenom)
+	mintAmount := k.CalcMintedCoins(ctx, total.Amount)
+	if mintAmount.IsZero() {
+		return mintedCoins, nil
+	}
+
+	// mint coins, update supply
+	mintedCoins = sdk.NewCoins(sdk.NewCoin(params.MintDenom, mintAmount.TruncateInt()))
+	err := k.bankKeeper.MintCoins(ctx, types.ModuleName, mintedCoins)
+	if err != nil {
+		return mintedCoins, err
+	}
+
+	// send the minted coins to the fee collector account
+	err = k.DistributeMintedCoin(ctx, mintedCoins)
+	if err != nil {
+		return mintedCoins, err
+	}
+
+	return mintedCoins, nil
+}
+
+func (k Keeper) CalcMintedCoins(ctx sdk.Context, totalAmt sdk.Int) sdk.Dec {
```

Comment:
i guess is better to change the use of sdk.int for  cosmossdk.io/math, since its deprecated